src/mono/mono/component/debugger-agent.c

```diff
@@ -1629,6 +1629,11 @@ mono_init_debugger_agent_for_wasm (int log_level_parm, MonoProfilerHandle *prof)
 	objrefs = g_hash_table_new_full (NULL, NULL, NULL, mono_debugger_free_objref);
 	obj_to_objref = g_hash_table_new (NULL, NULL);
 
+	/* Needed for the mono_g_hash_table_new_type_internal () call below */
+	mono_gc_base_init ();
+
+	suspended_objs = mono_g_hash_table_new_type_internal ((GHashFunc)mono_object_hash_internal, NULL, MONO_HASH_KEY_GC, MONO_ROOT_SOURCE_DEBUGGER, NULL, "Debugger Suspended Object Table");
+
```

Comment:
This seems like a copy of `objrefs_init` now (plus `mono_gc_base_init`). Maybe wasm should just call that?

Also, this doesn't seem needed for single-threaded wasm since `suspend_count` will never be greater than zero.  I'm surprised what the effect is on multi-threaded wasm.

Reply:
`objrefs_init` for some reason isn't used for WASM debugger (but used for WASI, I suppose for multi-threading). But the logic that causes asserts is the same for all types of a debugger agents. Is it intended?

Reply:
> `objrefs_init` for some reason isn't used for WASM debugger

I think this is probably unintentional - the WASM support was created in small pieces and it just initialized the parts that were used.  We should fix the duplicated logic.